### PR TITLE
Adding TrackArticleView to Admin View

### PR DESCRIPTION
**Pull request by aaronjae22**
The main function of this PR is to add the TrackArticleView model to Django Admin View and in this way be able to verify which articles were visited and how the user reached them.

image

Each visit to an article increments the corresponding count field displayed in the Admin View.

I'm currently working on a better way to filter.

### CODE REVIEW

File: portmap/core/admin.py [hunk 1]
```diff
@@ -61,3 +63,20 @@ class CustomUserAdmin(UserAdmin):
 @register(DataType, site=admin_site)
 class DataTypeAdmin(admin.ModelAdmin):
     list_display = ('id', 'name', 'helpText')
+
+@register(TrackArticleView, site=admin_site)
+class TrackArticleViewAdmin(admin.ModelAdmin):
+    list_display = ('article_path', "count", "visited_directly", "external_referrer",)
+    aggregated_counts = {}
+
+    def count(self, object):
```
Comment (lisad) [inline portmap/core/admin.py:72]: Cool isnt it??? This 'count' column would be even more appropriate added to the Article admin list display view, becuase then the sum number would show up once per Article.

**Review by lisad — APPROVED** [review]

File: portmap/core/admin.py [hunk 2]
```diff
@@ -61,3 +63,20 @@ class CustomUserAdmin(UserAdmin):
...
+        matched_item = next(item for item in self.aggregated_counts if item["article_path"] == object.article_path)
+        return matched_item["count"]
+
+    def get_queryset(self, request):
```
Comment (lisad) [inline portmap/core/admin.py:76]: You shouldn't need to do get_queryset or get_object here, especially if the count is moved to Article list display

**Review by lisad — APPROVED** [review]

File: portmap/core/admin.py (hunk 1 above)
Comment (aaronjae22) [reply portmap/core/admin.py:72]: That would be a better implementation, for sure.

**Comment by aaronjae22** [comment]
So, we decided to add the count in the Article admin view which seems far more useful. This meant to delete code in the `TrackArticleViewAdmin`.

We already had `def view_count` in the Article model so we modify it in order to return the count for each article.

**Review by lisad — APPROVED** [review]

**Pull request merged**